astroquery/mast/missions.py

```diff
@@ -0,0 +1,201 @@
+"""
+MAST Datasets
+=================
+
+This module contains methods for searching MAST datasets.
+"""
+
+import json
+
+import astropy.units as u
+import astropy.coordinates as coord
+
+from ..utils import commons, async_to_sync
+from ..utils.class_or_instance import class_or_instance
+
+from . import utils
+from .core import MastQueryWithLogin
+
+__all__ = ['DatasetsClass', 'Datasets']
+
+
+@async_to_sync
+class MissionsClass(MastQueryWithLogin):
+    """
+    Missions search class.
+
+    Class that allows direct programatic access to the MAST search API for a given mission.
+    """
+
+    def __init__(self):
+        super().__init__()
+        
+        self.service = 'search'
+        self.mission = 'hst'
+        
+        service_dict = {self.service: {'path': self.service, 'args': {}}}
+        self._service_api_connection.set_service_params(service_dict, f"{self.service}/{self.mission}")
+
+    def set_service(self, service):
+        """
+        Set the service name
+
+        Parameters
+        ----------
+        service : `str`
+            the name of the service
+        """
+        self.service = service
+
+    def set_mission(self, mission):
+        """
+        Set the mission name
+
+        Parameters
+        ----------
+        mission : `str`
+            the name of the mission
+        """
+
+        self.mission = mission
+
+    def _parse_result(self, response, verbose=False):  # Used by the async_to_sync decorator functionality
+        """
+        Parse the results of a `~requests.Response` objects and return an `~astropy.table.Table` of results.
+
+        Parameters
+        ----------
+        response : `~requests.Response`
+            `~requests.Response` objects.
+        verbose : bool
+            (presently does nothing - there is no output with verbose set to
+            True or False)
+            Default False.  Setting to True provides more extensive output.
+
+        Returns
+        -------
+        response : `~astropy.table.Table`
+        """
+
+        print(response)
+        return self._service_api_connection._parse_result(response, verbose, data_key='results')
+
+    @class_or_instance
+    def query_region_async(self, coordinates, radius=3*u.arcmin, **kwargs):
+        """
+        Given a sky position and radius, returns a list matching datasets.
+
+        Parameters
+        ----------
+        coordinates : str or `~astropy.coordinates` object
+            The target around which to search. It may be specified as a
+            string or as the appropriate `~astropy.coordinates` object.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 0.2 degrees.
+            The string must be parsable by `~astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `~astropy.units` may also be used. Defaults to 0.2 deg.
+        mission : str, optional
```

Comment:
I would make this a specific argument `mission="HST"` in the function signature, since it's such an important argument.

Also this argument is never used in the function, which it should be.